### PR TITLE
[CallWithChat] Updating the factory method for creating callwithchatadapter from stateful clients. 

### PR DESCRIPTION
**Pull request by JoshuaLai**
# What
Included new options bags in the factory method when creating from stateful client

# Why
Chat adapter needs the credential in the options bag to perform inline image download. 
So we need a new argument to be able to pass that downstream. 

# How Tested

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/communication-react/review/beta/communication-react.api.md [hunk 1]
```diff
@@ -182,6 +182,9 @@ export type AzureCommunicationCallWithChatAdapterFromClientArgs = {
     callClient: StatefulCallClient;
     chatClient: StatefulChatClient;
     chatThreadClient: ChatThreadClient;
+    options?: {
```
Comment (jpeng-ms) [inline packages/communication-react/review/beta/communication-react.api.md:185]: i think we might need to call out this change in storybook as well for `AzureCommunicationCallWithChatAdapterFromClientArgs`

**Review by jpeng-ms — APPROVED** [review]

File: packages/react-composites/review/beta/react-composites.api.md [hunk 2]
```diff
@@ -1039,7 +1042,9 @@ export const createAzureCommunicationCallWithChatAdapter: ({ userId, displayName
 export const _createAzureCommunicationCallWithChatAdapterFromAdapters: (callAdapter: CallAdapter, chatAdapter: ChatAdapter) => CallWithChatAdapter;
 
 // @public
-export const createAzureCommunicationCallWithChatAdapterFromClients: ({ callClient, callAgent, callLocator, chatClient, chatThreadClient }: AzureCommunicationCallWithChatAdapterFromClientArgs) => Promise<CallWithChatAdapter>;
+export function createAzureCommunicationCallWithChatAdapterFromClients(callClient: StatefulCallClient, callAgent: CallAgent, callLocator: CallAdapterLocator | TeamsMeetingLinkLocator, chatClient: StatefulChatClient, chatThreadClient: ChatThreadClient, options?: {
```
Comment (PorterNan) [inline packages/react-composites/review/beta/react-composites.api.md:1045]: Should we add this option to createAzureCommunicationCallWithChatAdapter as well? We should make last param as CallWithChatAdapterOptions instead

Then declare
type CallWithChatAdapterOption = CallAdapterOption && ChatAdapterOption

Comment (PorterNan) [reply packages/react-composites/review/beta/react-composites.api.md:1045]: How does inline image previously work in our callWithChat sample app?

Comment (JoshuaLai) [reply packages/react-composites/review/beta/react-composites.api.md:1045]: We only caught this issue when we just updated selfhost app yesterday. So it technically never worked / tested until today. 
We can for sure create a `type CallWithChatAdapterOption = CallAdapterOption && ChatAdapterOption` instead it might make the implementation a little cleaner. 

We shouldn't need to include it in `createAzureCommunicationCallWithChatAdapter` cause when we user that factory method the chat adapter grabs the credential when it creates the chat client itself and sends it down stream .

File: packages/communication-react/review/beta/communication-react.api.md (hunk 1 above)
Comment (JoshuaLai) [reply packages/communication-react/review/beta/communication-react.api.md:185]: Yes thats a good point. I think we created a new section in the storybook to include the factory methods for creating from stateful client, so this can be included in that new section.

File: packages/react-composites/review/beta/react-composites.api.md (hunk 2 above)
Comment (PorterNan) [reply packages/react-composites/review/beta/react-composites.api.md:1045]: Ohh right! Cause we pass in the token and create credential there

**Review by PorterNan — APPROVED** [review]

File: packages/react-composites/review/beta/react-composites.api.md [hunk 3]
```diff
@@ -810,6 +811,9 @@ export interface CallWithChatControlOptions extends CommonCallControlOptions {
 // @public
 export type CallWithChatEvent = 'callError' | 'chatError' | 'callEnded' | 'isMutedChanged' | 'callIdChanged' | 'isLocalScreenSharingActiveChanged' | 'displayNameChanged' | 'isSpeakingChanged' | 'callParticipantsJoined' | 'callParticipantsLeft' | 'selectedMicrophoneChanged' | 'selectedSpeakerChanged' | /* @conditional-compile-remove(close-captions) */ 'isCaptionsActiveChanged' | /* @conditional-compile-remove(close-captions) */ 'captionsReceived' | 'messageReceived' | 'messageSent' | 'messageRead' | 'chatParticipantsAdded' | 'chatParticipantsRemoved';
 
+// @beta
+export type CallWithChatOptions = ChatAdapterOptions;
```
Comment (JoshuaLai) [inline packages/react-composites/review/beta/react-composites.api.md:815]: @PorterNan So this is what it would look like if we introduced a new options. But since we don't have a `CallAdapterOptions` yet, it isn't a true union yet. So I don't know, when we do create a `CallAdapterOptions` will this be considered a breaking change?

File: packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts [hunk 4]
```diff
@@ -853,19 +863,25 @@ export type AzureCommunicationCallWithChatAdapterFromClientArgs = {
...
  * @public
  */
-export const createAzureCommunicationCallWithChatAdapterFromClients = async ({
-  callClient,
-  callAgent,
-  callLocator,
-  chatClient,
-  chatThreadClient
-}: AzureCommunicationCallWithChatAdapterFromClientArgs): Promise<CallWithChatAdapter> => {
+export async function createAzureCommunicationCallWithChatAdapterFromClients(
```
Comment (jimchou-dev) [inline packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts:866]: any reason why this change from const to function?

**Review by jimchou-dev — APPROVED** [review]

File: packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts (hunk 4 above)
Comment (JoshuaLai) [reply packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts:866]: The underlying functionality is the same, we just need to change it from an arrow function to just this normal function declaration cause it is the only way to include conditional compile flags in the arguments. This is a known limitation, we had to make this change in the ChatAdapter as well

File: packages/react-composites/review/beta/react-composites.api.md (hunk 3 above)
Comment (PorterNan) [reply packages/react-composites/review/beta/react-composites.api.md:815]: Just check the code, looks like we have AzureCommunicationCallAdapterOptions, I guess that is a good hint that we should align chatAdapter option name (a name change, we can do it in this PR)

And in the meantime, back to your question - if CallAdapterOption or AzureCommunicationCallAdapterOptions is not needed for now, let's directly use ChatAdapterOptions or AzureCommunicationChatAdapterOptions as the option type, then we can add

export type CallWithChatOptions = AzureCommunicationChatAdapterOptions && AzureCommunicationCallAdapterOptions; later when it is needed, changes like this is not considered as a breaking change

File: packages/react-composites/review/beta/react-composites.api.md [hunk 5]
```diff
@@ -1039,7 +1043,7 @@ export const createAzureCommunicationCallWithChatAdapter: ({ userId, displayName
 export const _createAzureCommunicationCallWithChatAdapterFromAdapters: (callAdapter: CallAdapter, chatAdapter: ChatAdapter) => CallWithChatAdapter;
 
 // @public
-export const createAzureCommunicationCallWithChatAdapterFromClients: ({ callClient, callAgent, callLocator, chatClient, chatThreadClient }: AzureCommunicationCallWithChatAdapterFromClientArgs) => Promise<CallWithChatAdapter>;
+export function createAzureCommunicationCallWithChatAdapterFromClients(callClient: StatefulCallClient, callAgent: CallAgent, callLocator: CallAdapterLocator | TeamsMeetingLinkLocator, chatClient: StatefulChatClient, chatThreadClient: ChatThreadClient, options?: CallWithChatOptions): Promise<CallWithChatAdapter>;
```
Comment (PorterNan) [inline packages/react-composites/review/beta/react-composites.api.md:1046]: The option part should be conditional compiled

Comment (JoshuaLai) [reply packages/react-composites/review/beta/react-composites.api.md:1046]: Wait are we sure? Won't that leak out into the stable api?

File: packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts [hunk 6]
```diff
@@ -832,6 +833,13 @@ export const useAzureCommunicationCallWithChatAdapter = (
   return adapter;
 };
 
+/* @conditional-compile-remove(teams-inline-images) */
+/**
+ * Configuration options to include when creating CallWithChatAdapter.
+ * @beta
+ */
+export type CallWithChatOptions = ChatAdapterOptions;
```
Comment (PorterNan) [inline packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts:841]: Again, if that is just the same type, we can just directly use ChatAdapterOptions, without introducing a new type here for now

Comment (PorterNan) [reply packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts:841]: Sry, AzureCommunicationChatAdapterOptions, to be more accurate

File: packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts [hunk 7]
```diff
@@ -832,6 +833,13 @@ export const useAzureCommunicationCallWithChatAdapter = (
   return adapter;
 };
 
+/* @conditional-compile-remove(teams-inline-images) */
```
Comment (PorterNan) [inline packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts:836]: Don't forget to remove this ;D

Comment (JoshuaLai) [reply packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts:836]: Oh good catch
